Client/Frontend/ContentBlocker/ContentBlockerHelper.swift

```diff
@@ -5,23 +5,118 @@
 import WebKit
 import Shared
 import Deferred
+import Data
+import BraveShared
 
 private let log = Logger.browserLogger
 
-enum BlocklistName: String {
-    case advertising = "disconnect-advertising"
-    case analytics = "disconnect-analytics"
-    case content = "disconnect-content"
-    case social = "disconnect-social"
-
-    var filename: String { return self.rawValue }
+// Rename to BlockList
+class BlocklistName: Equatable, Hashable {
+    static let ad = BlocklistName(filename: "block-ads")
+    static let tracker = BlocklistName(filename: "block-trackers")
+    static let https = BlocklistName(filename: "upgrade-http")
+    static let script = BlocklistName(filename: "block-scripts")
+    static let image = BlocklistName(filename: "block-images")
+
+    static var allLists: Set<BlocklistName> { return [.ad, .tracker, .https, .script, .image] }
+    
+    let filename: String
+    var rule: WKContentRuleList?
+    
+    init(filename: String) {
+        self.filename = filename
+    }
+    
+    static func blocklists(forDomain domain: Domain) -> (on: Set<BlocklistName>, off: Set<BlocklistName>) {
+        if domain.shield_allOff == 1 {
+            return ([], allLists)
+        }
+        
+        var onList = Set<BlocklistName>()
+        
+        if include(domainSetting: domain.shield_adblockAndTp, globalValue: Preferences.Shields.blockAdsAndTracking) {
+            onList.formUnion([.ad, .tracker])
+        }
+        
+        if include(domainSetting: domain.shield_noScript, globalValue: Preferences.Shields.blockScripts) {
+            onList.insert(.script)
+        }
+        
+        // TODO #159: Setup image shield
+        // TODO #269: Setup HTTPS shield
+        
+        return (onList, allLists.subtracting(onList))
+    }
+    
+    private static func include(domainSetting domain: NSNumber?, globalValue global: Preferences.Option<Bool>) -> Bool {
```

Comment:
Why not just make this an inline func in `blocklists(forDomain:)` since it's only called in there?